File: src/main/java/dev/emortal/minestom/marathon/util/BlockPacketUtils.java
package dev.emortal.minestom.marathon.util;

import net.minestom.server.adventure.audience.PacketGroupingAudience;
import net.minestom.server.coordinate.Point;
import net.minestom.server.network.packet.server.play.BlockBreakAnimationPacket;
import org.jetbrains.annotations.NotNull;

public interface BlockPacketUtils {
    static void sendBlockDamage(@NotNull PacketGroupingAudience audience, @NotNull Point point, byte destroyStage) {
        audience.sendGroupedPacket(new BlockBreakAnimationPacket(1, point, destroyStage));
    }
}

File: src/main/java/dev/emortal/minestom/marathon/util/BlockPacketUtils.kt
package dev.emortal.minestom.marathon.util

import net.minestom.server.adventure.audience.PacketGroupingAudience
import net.minestom.server.coordinate.Point
import net.minestom.server.instance.block.Block
import net.minestom.server.network.packet.server.play.BlockBreakAnimationPacket
import net.minestom.server.network.packet.server.play.EffectPacket

interface BlockPacketUtils {
    companion object {
        fun sendBlockDamage(audience: PacketGroupingAudience, point: Point, destroyStage: Byte) {
            audience.sendGroupedPacket(BlockBreakAnimationPacket(1, point, destroyStage))
        }

        fun sendBlockBreak(audience: PacketGroupingAudience, point: Point, block: Block) {
            audience.sendGroupedPacket(EffectPacket(2001, point, block.stateId(), false))
        }
    }
}
